docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,554 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e83ef718",
+   "metadata": {},
+   "source": [
+    "![MLRun Function](../_static/images/mlrun_function_diagram.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15c5c935",
+   "metadata": {},
+   "source": [
+    "Additionally, there are various parameters that are common across all functions including:\n",
+    "- **Name:** Name of the function (displayed in the UI and Kubernetes pod)\n",
+    "- **Tag:** Version of the function (can be manual or automatic)\n",
+    "- **Code:** What is actually being run (Python file, Jupyter notebook, Git repo, etc.)\n",
+    "- **Runtime:** How the code is being run (Job, Spark, Dask, Nuclio, etc.)\n",
+    "- **Image:** Docker image (or list of commands for building a new image)\n",
+    "- **Resources:** Memory, CPU, GPU resources. Also scaling/auto-scaling settings for supported runtimes (e.g. Nuclio, Serving, Horovod, etc.)\n",
+    "- **With Repo:** Whether the function depends on the MLRun project for additional files (e.g. Git repo, archive file, etc.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0933c707",
+   "metadata": {},
+   "source": [
+    "### Functions and Projects"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "040d4dd1",
+   "metadata": {},
+   "source": [
+    "The recommended way to create and run a function in MLRun is via a project. Once you register a function within a project, you can execute it in your local environment or at scale on a Kubernetes cluster."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41aeed94",
+   "metadata": {},
+   "source": [
+    "![MLRun Function](../_static/images/mlrun_concepts_architecture.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aa49c285",
+   "metadata": {},
+   "source": [
+    "An MLRun project object has a method called `set_function` which is a one-size-fits-all way of creating an MLRun function. The `set_function` method accepts a variety of sources including Python files, Jupyter notebooks, Git repos, and more.\n",
+    "\n",
+    "Additionally, the `set_function` method has a parameter called `with_repo` which denotes whether a function requires additional files or dependencies such as a Git repo or archive file. This Git repo or archive file is specified on a project level which the function will consume.\n",
+    "> Note: When using `with_repo`, the contents of the Git repo or archive file will be available in the current working directory of your MLRun function during runtime "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14800073",
+   "metadata": {},
+   "source": [
+    "## Creating a Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e2311f14",
+   "metadata": {},
+   "source": [
+    "There are 3 main scenarios for creating a function:\n",
+    "1. [Single Source File](#single-source-file) - when your code can be contained into a single file\n",
```

Comment:
"- [Single Source File](#single-source-file) - when your code can be contained into a single file\n",
    "- [Multiple Source Files](#multiple-source-files) - when your code requires additional files or dependencies\n",
    "- [Import Existing Function](#import-existing-function) - when your function already exists elsewhere and you just want to import it\n",
    "\n",
    "Using the `set_function` method of an MLRun project allows for each of these scenarios in a transparent way. Depending on the source passed in, the project registers the function using some lower level functions.\n",
    "> Note: For specific use cases, you also have access to these lower level functions. These are discussed in more detail at a later point.\n",
    
    Add reference!